mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -422,7 +433,11 @@ def read_metrics_data(
             is_empty=df.empty,
         )
 
-        return df_handler(df=df, metrics=metrics, project=self.project)
+        df_handler_kwargs["df"] = df
+
```

Comment:
```suggestion
if not mm_schemas.ResultData.RESULT_EXTRA_DATA not in df.columns and type="result":
     df[mm_schemas.ResultData.RESULT_EXTRA_DATA] = ""
```

If you will use my suggestion, you can remove the new parameter from `df_to_results_values` methoods and revert the df_handler_kwargs changes, which I believe will help simplify the code.

of course it can be done in v3io connector as well